src/utils/projectV2.ts

```diff
@@ -44,22 +34,40 @@ const paramsToDelete = [
  * @returns An updated query object with preserved, removed, and new parameter
  */
 
-export const updateUrlWithParams = (
+export const buildProjectDetailsQuery = (
   asPath: string,
   query: ParsedUrlQuery,
-  siteId: string | null
-) => {
-  const [, queryString] = asPath.split('?');
-  const currentUrlParams = new URLSearchParams(queryString || '');
-  const currentQuery = { ...query };
-  paramsToPreserve.forEach((param) => {
-    if (currentUrlParams.has(param)) {
-      currentQuery[param] = currentUrlParams.get(param) ?? '';
-    }
-  });
+  routeParams: RouteParams
+): Record<string, string> => {
+  console.log('updateUrlWithParams:');
+  console.log('  Path:', asPath);
+  console.log('  Query:', query);
+  console.log('  Route Params:', routeParams);
+  // Convert ParsedUrlQuery to Record<string, string> by filtering out non-string values
+  const currentQuery: Record<string, string> = Object.entries(query).reduce(
+    (stringQueryParams, [key, value]) => {
+      if (typeof value === 'string') {
+        stringQueryParams[key] = value;
+      }
+      return stringQueryParams;
+    },
+    {} as Record<string, string>
+  );
+
```

Comment:
To improve readability, can we refactor the reduce method? We can use the logic below to achieve the same result.
```
const currentQuery: Record<string, string> = Object.fromEntries(
  Object.entries(query).filter(([_, value]) => typeof value === 'string')
);
```

Reply:
Done, had to add an additional type check.